### PR TITLE
Adding more Elements

### PR DESCRIPTION
**Pull request by Arraxx**
# Description

Added more elements with the reference below, excluding the elements present in PR #5451 

Reference:
https://pubchem.ncbi.nlm.nih.gov/
https://www.rsc.org/periodic-table/
https://www.ciaaw.org/

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by poshul** [comment]
How are abundances handled for some of the heavier radioactive elements? Plutonium for example seems to have 0 abundances for all the isotopes. (Though if I ever run into organo-plutonium biomolecules I'd probably change jobs to something safer...)

**Comment by Arraxx** [comment]
@poshul  as I was not able to find out the abundance of some elements that's why I added zero for them but on further searching i found that it can't be zero, so it will be great if you can suggest me something suitable for this.(or we can take any variable number)

**Comment by jpfeuffer** [comment]
We have multiple options for non-natural elements:
- distribute equally
- use 1.0 at the first isotope
- fail/warn when its average weight is used

It does not matter much as long as it is documented well, since the use cases should be very rare. If someone uses them, they need to use the actual isotopes in their formulas anyway, since they most likely are looking to distinguish and quantify the different isotopes separately.

**Comment by poshul** [comment]
I think we should leave out anything that is not naturally occurring. The isotope abundance is going to be dependent on how and where it was made. That applies to anything Transuranic, as well as Promethium. This leaves us with Po At Rn Fr and Ra, as naturally occurring elements that I cant find isotope abundance for. I'm open to suggestions as to handling these five.

Also I added Protactinium and Uranium.

**Comment by jpfeuffer** [comment]
Polonium is here: https://www.chemie.de/lexikon/Polonium.html
You can check for others as well. Table on the right side (NH)

**Comment by Arraxx** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
looks good. does anyone know if relative abundacnes need to add up to 1.0 ? e.g., what if the sum deviates too much?

**Comment by Arraxx** [comment]
any suggestions? @jpfeuffer @poshul

**Comment by jpfeuffer** [comment]
@timosachsenberg most probably. How else would you calculate correct average masses? Unless we normalize anyway, but then it also yields wrong results if it was supposed to add up to 1 but doesn't.

**Comment by jpfeuffer** [comment]
My preferred solution is failing/warning. So maybe we should remove/not add elements that do not have known abundances (although potentially monoisotopic mass could be used).
And document in the class documentation where we got the abundances from and what to do when an Element ist missing.

**Comment by timosachsenberg** [comment]
I will check if we normalize after reading

**Comment by timosachsenberg** [comment]
So what action should we implement?
In https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/CHEMISTRY/ElementDB.cpp#L607
add up abundances and add an OPENMS_POSTCONDITION that sum of abundances is >= 0.95 ?

**Comment by poshul** [comment]
I'd say setting an OPENMS_POSTCONDITION is the way to handle it, and just not include elements without abundances.

**Comment by timosachsenberg** [comment]
@Arraxx can you take on that task?

**Comment by Arraxx** [comment]
okay @timosachsenberg

**Review by timosachsenberg — APPROVED** [review]

**Comment by timosachsenberg** [comment]
I think not all are addressed yet?

> So what action should we implement?
> In https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/CHEMISTRY/ElementDB.cpp#L607
> 
> * add up abundances and add an OPENMS_POSTCONDITION that sum of abundances is >= 0.95 ?
> * document in class documentation where abundances are from
> * add a sentence to class doc "Note: if an element is missing, you can add it manually or open an issue on the OpenMS repository."

**Comment by timosachsenberg** [comment]
@Arraxx do you plan to move forward here?

**Comment by Arraxx** [comment]
Sure, I will try to complete it. @timosachsenberg

**Comment by Arraxx** [comment]
@timosachsenberg , can you please explain it a little more to me.

**Comment by jpfeuffer** [comment]
What is the status here? We have a usecase and it is getting urgent.

**Comment by timosachsenberg** [comment]
I checked iridium and osmium. And it is safe to add them because the abundances add up to 1.0
For the other ones we are not sure if this has any negative impact (e.g., if total abundances are zero but we normalize internally through the sum of abundances)

**Comment by timosachsenberg** [comment]
To get something actionable: I would suggest that we finish the PR with elements with abundances that add up to 1.0 and look into the others later.

**Comment by jpfeuffer** [comment]
I would say we just fail if someone wants an average weight for this element.

**Comment by jpfeuffer** [comment]
Creating compounds with a specific isotope would at least work.

**Comment by timosachsenberg** [comment]
I think it is not only the average weight but a lot of other things (e.g., coarse isotope patterns)

**Comment by jpfeuffer** [comment]
Then this fails, too. And the user is forced to provide a specific isotope.

**Comment by timosachsenberg** [comment]
I removed the recently added ones that have zero abundance.
Still would be good to revisit them. E.g., adding them to enable e.g. monoisotopic mass calculation but failing if an unsupported operation is performed on them.

**Comment by timosachsenberg** [comment]
rebuild jenkins

File: src/openms/source/CHEMISTRY/ElementDB.cpp [hunk 1]
```diff
@@ -496,8 +497,8 @@ namespace OpenMS
     buildElement_("Hafnium", "Hf", 72u, hafnium_abundance, hafnium_mass);
 
 
-    map<unsigned int, double> tantalum_abundance = {{181u, 1.0}};
-    map<unsigned int, double> tantalum_mass = {{181u, 180.947995800000001}};
+    map<unsigned int, double> tantalum_abundance = {{180u, 0.0001176}, {181u, 1.0}};
```
Comment (hroest) [inline src/openms/source/CHEMISTRY/ElementDB.cpp:500]: these should add to 100%, no?

**Comment by dlon450** [comment]
Hey, just wondering what factors are involved in the process for adding more (complicated) elements? I've got a list of missing elements that would help a lot for the tool I'm currently working on. I'm no chemistry expert but would be happy to provide further details/information and ask the chemists I'm working with how we can be of help. Thanks
[missing_elements.csv](https://github.com/OpenMS/OpenMS/files/8189879/missing_elements.csv)

**Comment by hroest** [comment]
@dlon450 maybe we can have a discussion on discord, the main issues are conceptual as the current OpenMS only considers stable isotopes. We currently simply do not have proper modelling for non-stable isotopes that are not naturally occurring at a stable frequency. We would need to somehow incorporate radioactive isotopes, potentially with a modelling half lives which requires some discussion. 

Therefore, the process is a discussion with a proposal of proper modelling and then a PR

**Comment by dlon450** [comment]
@hroest yes that would be great. My discord is
Derek!!#6699 
so maybe you can add me first and we can discuss this more there. Thanks

**Comment by hroest** [comment]
I added you on Discord -- or just join us here at https://discord.gg/4TAGhqJ7s5

**Comment by hroest** [comment]
I merged this with f3910ff6988b4910ced0787a964a8e48ced92

**Pull request merged**